src/mahoji/lib/abstracted_commands/camdozaalCommand.ts

```diff
@@ -0,0 +1,174 @@
+	let [duration, newQuantity] = determineMiningTime({
+		quantity,
+		user,
+		ore: barroniteRocks,
+		ticksBetweenRolls: currentPickaxe.ticksBetweenRolls,
+		glovesRate,
+		armourEffect,
+		miningCapeEffect,
+		powermining: powermine,
+		goldSilverBoost,
+		miningLvl: miningLevel
+	});
+
+	const fakeDurationMin = quantity ? randomVariation(reduceNumByPercent(duration, 25), 20) : duration;
+	const fakeDurationMax = quantity ? randomVariation(increaseNumByPercent(duration, 25), 20) : duration;
+
+	await addSubTaskToActivityTask<ActivityTaskOptionsWithQuantity>({
+		userID: user.id,
+		channelID: channelID.toString(),
+		quantity: newQuantity,
+		duration,
+		type: 'CamdozaalMining'
+	});
+
+	let response = `${minionName(user)} is now mining inside the Ruins of Camdozaal until your minion ${
+		quantity ? `mined ${quantity}x barronite rocks or gets tired` : 'is satisfied'
+	}, it'll take ${
+		quantity
+			? `between ${formatDuration(fakeDurationMin)} **and** ${formatDuration(fakeDurationMax)}`
+			: formatDuration(duration)
+	} to finish.`;
+
+	if (boosts.length > 0) {
+		response += `\n\n**Boosts:** ${boosts.join(', ')}.`;
+	}
+
+	return response;
+}
+
+async function smithingCommand(user: MUser, channelID: string, quantity: number | undefined) {
+	if (user.skillLevel(SkillsEnum.Smithing) < 14) {
+		return 'You need at least level 14 Smithing to smith in the Ruins of Camdozaal';
+	}
+
+	const maxTripLength = calcMaxTripLength(user, 'CamdozaalSmithing');
+	const timePerSmith = 3.5 * Time.Second;
+	if (!quantity) {
+		quantity = Math.floor(maxTripLength / timePerSmith);
+	}
+
+	const itemsNeeded = new Bank({ 'Barronite deposit': 1 }).clone();
+	const maxCanDo = user.bank.fits(itemsNeeded);
+	if (maxCanDo === 0) {
+		return "You don't own a single Barronite deposit!";
+	}
+	if (maxCanDo < quantity) {
+		quantity = maxCanDo;
+	}
+
+	const duration = timePerSmith * quantity;
+	if (duration > maxTripLength) {
+		return `${user.minionName} can't go on trips longer than ${formatDuration(
+			maxTripLength
+		)}, try a lower quantity. The highest amount of Barronite deposits you can smith is ${Math.floor(
+			maxTripLength / timePerSmith
+		)}.`;
+	}
+
+	await user.removeItemsFromBank(new Bank().add('Barronite deposit', quantity));
+
+	await addSubTaskToActivityTask<ActivityTaskOptionsWithQuantity>({
+		userID: user.id,
+		channelID: channelID.toString(),
+		quantity,
+		duration,
+		type: 'CamdozaalSmithing'
+	});
+
+	return `${user.minionName} is now smithing in the Ruins of Camdozaal, it will take around ${formatDuration(
+		duration
+	)} to finish.`;
+}
+
+async function fishingCommand(user: MUser, channelID: string, quantity: number | undefined) {
+	if (user.skillLevel(SkillsEnum.Fishing) < 7) {
+		return 'You need at least level 7 Fishing to fish in the Ruins of Camdozaal';
+	}
+
+	const maxTripLength = calcMaxTripLength(user, 'CamdozaalFishing');
+	const timePerFish = 5.5 * Time.Second;
```

Comment:
Time per fish here is 5.5 but in the fish declaration it is 1. (consistency)

Reply:
the timePerFish is just a placeholder for Fish[] and isn't used.

Reply:
I can look into adding something that uses it properly though

Reply:
reworked